### PR TITLE
Add module for Draggable gadget

### PR DESCRIPTION
**Pull request by laurenrad**
Adding a module to implement the Toolbox's Draggable gadget.

### CODE REVIEW

File: riscos_toolbox/gadgets/draggable.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	_fields_ = [ ("text", ctypes.c_char_p),
+	             ("max_text_len", ctypes.c_int32),
+	             ("sprite", ctypes.c_char_p),
+	             ("max_sprite_len", ctypes.c_int32) ]
+	            
+class DraggableDragStartedEvent(ToolboxEvent):
+	event_id = Draggable.DragStarted
+	
+	# This event actually doesn't have any fields other than the header
+	_fields_ = [ ]
```
Comment (c-jo) [inline riscos_toolbox/gadgets/draggable.py:68]: If there aren't any more fields you don't strictly need this.

File: riscos_toolbox/gadgets/draggable.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	
+class DraggableDragEndedEvent(ToolboxEvent):
+	event_id = Draggable.DragEnded
+	             	
+	 # Note that window handle and icon handle here can be object/component ID if
+	 # ToolboxIds flag is set
```
Comment (c-jo) [inline riscos_toolbox/gadgets/draggable.py:74]: Maybe add a property toolbox_id which returns a tuple of the object/component ID. If you want to be clever the state of ToolboxIds could be checked in this (and window_handle/icon_handle) and raise an exception if you use the "wrong" one.

**Review by c-jo — APPROVED** [review]
A few small comments.

**Comment by c-jo** [comment]
The C header file also defines these which I think are flags for the DragEndedEvent.

Could they be a set of proprties on the event?

#define Draggable_DragStarted_Adjust  0x00000001
#define Draggable_DragStarted_Menu    0x00000002
#define Draggable_DragStarted_Select  0x00000004
#define Draggable_DragStarted_Shift   0x00000008
#define Draggable_DragStarted_Ctrl    0x00000010

File: riscos_toolbox/gadgets/draggable.py (hunk 1 above)
Comment (laurenrad) [inline riscos_toolbox/gadgets/draggable.py:68]:
```suggestion
```

File: riscos_toolbox/gadgets/draggable.py (hunk 2 above)
Comment (laurenrad) [reply riscos_toolbox/gadgets/draggable.py:74]: I'm not quite sure what you mean by the second point as far as using the wrong one, but on the first one, the window_handle and icon_handle can be object/component id if the ToolboxIds flag is set, so perhaps we could make this a single tuple property called more generically "ids" or something, and provide some kind of method in the Draggable class to easily flip that flag using the Gadget flags setter? Does that make sense?

**Pull request merged**